ldi-orchestrator/ldio-connectors/ldio-archive-file-in/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioArchiveFileInAutoConfig.java

```diff
@@ -0,0 +1,52 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldio.config;
+
+import be.vlaanderen.informatievlaanderen.ldes.ldi.services.ComponentExecutor;
+import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiAdapter;
+import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiComponent;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.ArchiveFileCrawler;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.LdioArchiveFileIn;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.LdioArchiveFileInRunner;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.configurator.LdioInputConfigurator;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.valueobjects.ComponentProperties;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFLanguages;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Configuration
+public class LdioArchiveFileInAutoConfig {
+
+	public static final String ARCHIVE_ROOT_DIR_PROP = "archive-root-dir";
+	public static final String SOURCE_FORMAT = "source-format";
+
+	@Bean("be.vlaanderen.informatievlaanderen.ldes.ldio.LdioArchiveFileIn")
+	public LdioInputConfigurator ldiArchiveFileInConfigurator() {
+		return new LdioInputConfigurator() {
+			@Override
+			public LdiComponent configure(LdiAdapter adapter,
+					ComponentExecutor executor,
+					ComponentProperties config) {
+				ArchiveFileCrawler archiveFileCrawler = new ArchiveFileCrawler(getArchiveDirectoryPath(config));
+				Lang sourceFormat = getSourceFormat(config);
+				var ldioArchiveFileInRunner = new LdioArchiveFileInRunner(executor, archiveFileCrawler, sourceFormat);
+				return new LdioArchiveFileIn(executor, ldioArchiveFileInRunner);
+			}
+
+			private Path getArchiveDirectoryPath(ComponentProperties config) {
+				String directory = config.getProperty(ARCHIVE_ROOT_DIR_PROP);
+				return Paths.get(directory);
+			}
+
+			private Lang getSourceFormat(ComponentProperties config) {
+				return config.getOptionalProperty(SOURCE_FORMAT)
+						.map(RDFLanguages::nameToLang)
+						.orElse(null);
```

Comment:
would expect a default if it's an optional property

Reply:
This is only used as a hint by the RDFParser but it might be a good idea to make the default jsonLD as .nq and .ttl can easily be recognized by jena but .json cannot be parsed without hint.

Reply:
This may be confusing in the docs as this is a 'hint' language. If jena cannot deduce the file format, it will use this hint. If I document that the default is TURTLE, then people using .nq files will think that this property must be included with NQ for it to work, which is not true. It is for cases like this that @Nullable and @NotNull annotations would be valuable ;). Then you can safely pass a null value.